### PR TITLE
fix: new conversation search fix

### PR DESCRIPTION
**Pull request by MohammedMokresh**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

after adding the new group name screen the search flow was not working properly

### Solutions

update the navigation flow by using nested navhosts

#### How to Test

go to the new conversation flow and try to search 

## before
Wire 2022-04-11 at 6_33 PM

## after 
new convo flow

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationRouter.kt
Comment (trOnk12) [inline app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationRouter.kt:15]: not sure about the names here :P maybe listNavController = searchNavController and navControlller = newConversationNavController or something like that ?

**Review by trOnk12 — APPROVED** [review]
lgtm, if you think my suggestion about naming is bullshit, ignore it and merge it, otherwise you can change it before merge

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-501/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3079404)(3079404)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3079404)(3079404)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationRouter.kt
Comment (MohammedMokresh) [reply app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationRouter.kt:15]: yeah I think it make more sense , will update it

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-501/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3079666)(3079666)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3079666)(3079666)-dev-release.apk)

**Pull request merged**
